### PR TITLE
Refactor of onekey to support multiple development boards 

### PR DESCRIPTION
**Pull request by zvecr**
## Description

Added a structure which provides reference examples for common boards. This can help with diagnosis of issues, and should help to clear up some of the recent-ish questions.

This removes the need to have `keyboards/chibios_test/teensy_lc_onekey` (which gives a non complete example of teensy 3.2 support), maybe `keyboards/proton_c` as well as underpin simplification work such as #5669.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/onekey/elite_c/rules.mk [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   atmega32a    bootloadHID
+BOOTLOADER = dfu
```
Comment (drashna) [inline keyboards/handwired/onekey/elite_c/rules.mk]: Just `dfu` doesn't actually do anything.  This needs to be explicitly set.  
```suggestion
BOOTLOADER = atmel-dfu
```

(`bootloader.mk` is where the magic sauce is)

**Comment by zvecr** [comment]
Requires #6035 for some of the documented pins.

File: keyboards/handwired/onekey/elite_c/rules.mk (hunk 1 above)
Comment (zvecr) [reply keyboards/handwired/onekey/elite_c/rules.mk]: Darn missed that one. Build didn't produce warnings about unknown bootloader but that's a PR for another day.

**Comment by zvecr** [comment]
#6035 has been merged now and this has been rebased. Should be good to go now.

**Comment by drashna** [comment]
Thanks!

**Pull request merged**